cmake/TestImpactFramework/LYTestImpactFramework.cmake

```diff
@@ -324,31 +324,64 @@ function(ly_test_impact_write_gem_target_enumeration_file GEM_TARGET_TEMPLATE_FI
      configure_file(${GEM_TARGET_TEMPLATE_FILE} ${mapping_path})
 endfunction()
 
+#! ly_extract_aliased_target_dependencies: recursively extracts the aliases of a target to retrieve the true de-aliased target.
+#
+# \arg:TARGET target to de-alias
+function(ly_extract_aliased_target_dependencies TARGET)
+    if(ARGN STREQUAL "")
+        # Entry point of recursive call, set the parent target and clear any existing aliases
+        set(PARENT_TARGET ${TARGET})
+        set_property(GLOBAL PROPERTY LY_DE_ALIASED_TARGETS_${PARENT_TARGET} "")
+    endif()
+
+    # Check for aliases of this target
+    get_property(aliased_targets GLOBAL PROPERTY LY_ALIASED_TARGETS_${TARGET} SET)
+    if(${aliased_targets})
+        # One or more aliases for this target has been found
+        get_property(aliased_targets GLOBAL PROPERTY LY_ALIASED_TARGETS_${TARGET})
+        foreach(aliased_target ${aliased_targets})
+            # Recursively extract any aliases of the alias of this target
+            ly_extract_aliased_target_dependencies(${aliased_target} ${PARENT_TARGET})
+        endforeach()
+    else()
+        # No more aliases found for this target, add this target as an alias for the parent target
+        set_property(GLOBAL APPEND PROPERTY LY_DE_ALIASED_TARGETS_${PARENT_TARGET} ${TARGET})
+    endif()
+endfunction()
+
 #! ly_extract_target_dependencies: extracts the target dependencies for the specified target as a comma separated list.
 function(ly_extract_target_dependencies INPUT_DEPENDENCY_LIST OUTPUT_DEPENDENCY_LIST)
     set(dependencies "")
     # Walk the dependency list
     foreach(target_name_components ${INPUT_DEPENDENCY_LIST})
         # Extract just the target name, ignoring the namespace
-        string(REPLACE "::" ";" target_name_components ${target_name_components})
-        list(LENGTH target_name_components num_name_components)
-        if(num_name_components GREATER 1)
-            list(GET target_name_components 0 target_namespace)
-            list(GET target_name_components 1 target_name)
-            # Skipt third party dependencies
-            if(NOT target_namespace STREQUAL "3rdParty")
-                list(APPEND dependencies "\"${target_name}\"")
+        if(TARGET ${target_name_components})
+            set(target_to_add "")
+            string(REPLACE "::" ";" target_name_components ${target_name_components})
+            list(LENGTH target_name_components num_name_components)
+            if(num_name_components GREATER 1)
+                list(GET target_name_components 0 target_namespace)
+                list(GET target_name_components 1 target_name)
+                # Skipt third party dependencies
+                if(NOT target_namespace STREQUAL "3rdParty")
+                    set(target_to_add ${target_name})
+                endif()
```

Comment:
This isn't the correct way to filter out 3rdParty libraries.
Search for a target that starts with "3rdParty" doesn't mean that targets that doesn't start with "3rdParty" aren't not 3rdParty libraries.

The proper way to detect 3rdParty libraries can be seen in [RuntimeDependencies_Common.cmake](https://github.com/o3de/o3de/commit/7c0e902a8a01f652345a29be68050d5bba52f4ec).

What should be check is if the TARGET has an IMPORTED property, which means it is not built by O3DE.

Reply:
The only problem with checking for `IMPORTED` is that target aliases are also tagged as `IMPORTED`, therefore a simple check for this property on a given target will only tell us that the target is either 3rd party _or_ an alias. As it stands, there does not appear a conclusive way of diffrentiating between 3rd party targets and aliased targets, as if a 3rd party target is aliased, then simply checking whether or not a target is aliased is not going to be enough to make this distinction.

Consider the fully qualified target names with the `IMPORT` property set:

```
3rdParty::AWSGameLiftServerSDK
3rdParty::AWSNativeSDK::AWSClientAuth
3rdParty::AWSNativeSDK::AWSCore
3rdParty::AWSNativeSDK::Core
3rdParty::AWSNativeSDK::GameLiftClient
3rdParty::Blast
3rdParty::Crashpad
3rdParty::Crashpad::Handler
3rdParty::DirectXShaderCompilerDxc
3rdParty::Freetype
3rdParty::GoogleBenchmark
3rdParty::ISPCTexComp
3rdParty::Lua
3rdParty::NvCloth
3rdParty::OpenEXR
3rdParty::OpenGLInterface
3rdParty::OpenImageIO
3rdParty::OpenImageIO::Tools::Binaries
3rdParty::OpenImageIO::Tools::PythonPlugins
3rdParty::OpenSSL
3rdParty::PNG
3rdParty::PhysX
3rdParty::Python
3rdParty::Qt::Concurrent
3rdParty::Qt::Core
3rdParty::Qt::Gui
3rdParty::Qt::Network
3rdParty::Qt::OpenGL
3rdParty::Qt::Svg
3rdParty::Qt::Test
3rdParty::Qt::Widgets
3rdParty::Qt::Xml
3rdParty::RapidJSON
3rdParty::RapidXML
3rdParty::SPIRVCross
3rdParty::SQLite
3rdParty::TIFF
3rdParty::ZLIB
3rdParty::assimplib
3rdParty::astc-encoder
3rdParty::azslc
3rdParty::cityhash
3rdParty::d3dx12
3rdParty::expat
3rdParty::glad_vulkan
3rdParty::googletest::GMock
3rdParty::googletest::GTest
3rdParty::libsamplerate
3rdParty::lz4
3rdParty::mcpp
3rdParty::md5
3rdParty::mikkelsen
3rdParty::poly2tri
3rdParty::pybind11
3rdParty::pyside2
3rdParty::squish-ccr
3rdParty::v-hacd
3rdParty::vulkan-validationlayers
3rdParty::xxhash
3rdParty::zstd
Gem::AWSCore.Clients
Gem::LyShine.Clients.API
Gem::MaterialCanvas.Builders
Gem::MaterialEditor.Builders
Gem::ShaderManagementConsole.Builders
Gem::UiBasics.Builders
Gem::UiBasics.Tools
```

Here we can see that the targets not namespaced as `3rdParty` are our aliased targets, but you are correct in affirming that we cannot rely on the `3rdParty` namespace to conclusively determine whether not a given target is a 3rd party import. However, for now there does not appear to be an elegant solution to this problem so I will push the changes as-is and I have opened up a GHI to investigate this issue further to determine a way of unambiguously differentiating between 3rd party targets and non-3rd party aliased targets here: https://github.com/o3de/o3de/issues/11280